src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs

```diff
@@ -146,31 +148,70 @@ internal static bool ValidateSslServerCertificate(string targetServerName, X509C
                     return true;
                 }
 
-                if ((policyErrors & SslPolicyErrors.RemoteCertificateNameMismatch) != 0)
+                if (policyErrors.HasFlag(SslPolicyErrors.RemoteCertificateChainErrors))
                 {
+                    SqlClientEventSource.Log.TrySNITraceEvent(nameof(SNICommon), EventType.ERR, "targetServerName {0}, SslPolicyError {1}, SSL Policy certificate chain has errors.", args0: targetServerName, args1: policyErrors);
+
+                    // get the chain status from the certificate
+                    X509Certificate2 cert2 = cert as X509Certificate2;
+                    X509Chain chain = new();
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.Offline;
+                    StringBuilder chainStatusInformation = new();
+                    bool chainIsValid = chain.Build(cert2);
+                    if (!chainIsValid)
+                    {
+                        foreach (X509ChainStatus chainStatus in chain.ChainStatus)
+                        {
+                            if (chainStatusInformation.Length == 0)
+                            {
+                                chainStatusInformation.Append(Environment.NewLine);
+                            }
+                            chainStatusInformation.Append(chainStatus.StatusInformation);
```

Comment:
Can you include the `Status` enum in the `chainStatusInformation` as well?
```suggestion
                            chainStatusInformation.Append(chainStatus.StatusInformation);
```